scripts/setup-centos9.sh

```diff
@@ -31,18 +31,28 @@ set -efx -o pipefail
 SCRIPTDIR=$(dirname "${BASH_SOURCE[0]}")
 source $SCRIPTDIR/setup-helper-functions.sh
 NPROC=$(getconf _NPROCESSORS_ONLN)
-export CFLAGS=$(get_cxx_flags)  # Used by LZO.
-export CXXFLAGS=$CFLAGS  # Used by boost.
-export CPPFLAGS=$CFLAGS  # Used by LZO.
+export CXXFLAGS=$(get_cxx_flags) # Used by boost. 
+export CFLAGS=${CXXFLAGS//"-std=c++17"/} # Used by LZO.
 CMAKE_BUILD_TYPE="${BUILD_TYPE:-Release}"
 BUILD_DUCKDB="${BUILD_DUCKDB:-true}"
-export CC=/opt/rh/gcc-toolset-12/root/bin/gcc
-export CXX=/opt/rh/gcc-toolset-12/root/bin/g++
+USE_CLANG="${USE_CLANG:-false}"
 
 function dnf_install {
   dnf install -y -q --setopt=install_weak_deps=False "$@"
 }
 
+function install_and_setup_clang15 {
+  dnf_install clang15 gcc-toolset-13-libatomic-devel
+  CLANG_NUM_ALTERNATIVES=`alternatives --list | grep "clang " | wc -l`
+  if [[ ${CLANG_NUM_ALTERNATIVES} -eq 0 ]]; then
+    alternatives --install /usr/bin/clang clang /usr/bin/clang-15 80 --follower /usr/bin/clang++ clang++ /usr/bin/clang++-15
+  fi
+  (test ! -h /usr/local/bin/clang && ln -s /usr/bin/ccache /usr/local/bin/clang) || true
```

Comment:
Why are we symlinking ccache to clang ?

Reply:
The build system uses ccache to invoke the selected compiler. This linking ensures ccache is aware of clang (like it is of gcc which is linked by default). Because we are installing a non-system default version of clang (clang15) ccache doesn't get the clang links automatically added (because it would conflict if the system default clang - on Centos9 now clang18) is installed.

As an example, this is what it looks like on my Centos9 build system (with having run this setup script):
```
[root@c38594v1 ccache]# pwd
/usr/lib64/ccache
[root@c38594v1 ccache]# ls -ltr
total 0
lrwxrwxrwx 1 root root 16 Mar 29 11:59 cc -> ../../bin/ccache
lrwxrwxrwx 1 root root 16 Mar 29 11:59 gcc -> ../../bin/ccache
lrwxrwxrwx 1 root root 16 Mar 29 11:59 x86_64-redhat-linux-gcc -> ../../bin/ccache
lrwxrwxrwx 1 root root 16 Mar 29 11:59 c++ -> ../../bin/ccache
lrwxrwxrwx 1 root root 16 Mar 29 11:59 x86_64-redhat-linux-c++ -> ../../bin/ccache
lrwxrwxrwx 1 root root 16 Mar 29 11:59 g++ -> ../../bin/ccache
lrwxrwxrwx 1 root root 16 Mar 29 11:59 x86_64-redhat-linux-g++ -> ../../bin/ccache
lrwxrwxrwx 1 root root 16 Jul  4 12:20 clang -> ../../bin/ccache
lrwxrwxrwx 1 root root 16 Jul  4 12:20 clang++ -> ../../bin/ccache
```

Reply:
The above mechanism is supported by ccache but they also say https://ccache.dev/manual/4.2.html#_using_ccache_with_other_compiler_wrappers that we can accomplish this by setting the prefix command ? Can you try that out ? That might make things more explicit and easier to test imo.

Reply:
Setting the CC and CXX variables prior to the build is enough. I removed the ccache symlinks and CMake will use the ccache prefixes properly.

Reply:
The only issue is boost. Without the alternative (that I removed) it won't accept toolset clang because it looks for clang/clang++. So it will use the guessed toolset (if gcc is present it seems to use that) or a newer/different clang version. Perhaps if this is a problem we can revisit this. I added a comment for boost.